ams/ams-server/src/main/java/com/netease/arctic/ams/server/utils/CatalogUtil.java

```diff
@@ -98,11 +102,19 @@ public static Set<TableIdentifier> loadTablesFromCatalog() {
     Set<TableIdentifier> tables = new HashSet<>();
     List<CatalogMeta> catalogMetas = ServiceContainer.getCatalogMetadataService().getCatalogs();
     catalogMetas.forEach(catalogMeta -> {
-      ArcticCatalog arcticCatalog =
-          CatalogLoader.load(ServiceContainer.getTableMetastoreHandler(), catalogMeta.getCatalogName());
-      List<String> databases = arcticCatalog.listDatabases();
-      for (String database : databases) {
-        tables.addAll(arcticCatalog.listTables(database));
+      try {
+        ArcticCatalog arcticCatalog =
+            CatalogLoader.load(ServiceContainer.getTableMetastoreHandler(), catalogMeta.getCatalogName());
+        List<String> databases = arcticCatalog.listDatabases();
+        for (String database : databases) {
+          try {
+            tables.addAll(arcticCatalog.listTables(database));
+          } catch (Exception e1) {
+            LOG.error("list tables for database {} error", database, e1);
+          }
+        }
+      } catch (Exception e2) {
+        LOG.error("load table for catalog {} error", catalogMeta.getCatalogName(), e2);
```

Comment:
Catching the exception (and ignoring it) will bring risk: when failing to load the catalog/database, we will delete all the cached tables in this catalog/database, which means we will delete all the optimize `runtime` `tasks` and` history`